File: src/main/java/model/pieces/Queen.java
package model.pieces;

import model.moves.Move;
import model.util.Colour;
import model.util.Offset;

public class Queen extends DirectionPiece {
    private static final Offset[] OFFSETS = {
            Offset.TOP_LEFT,
            Offset.UP,
            Offset.TOP_RIGHT,
            Offset.LEFT,
            Offset.RIGHT,
            Offset.BOTTOM_LEFT,
            Offset.DOWN,
            Offset.BOTTOM_RIGHT
    };

    public Queen(Colour colour) {
        super(colour);
    }

    @Override
    Offset[] getDirections() {
        return OFFSETS;
    }

    @Override
    int getUnicodeWhite() {
        return 9813;
    }

    @Override
    int getUnicodeBlack() {
        return 9819;
    }

    @Override
    public int getUnsignedValue() {
        return 8;
    }

    @Override
    public void notifyMoveComplete(Move move) {
    }

    @Override
    public void notifyMoveUndo(Move move) {
    }

    @Override
    String getName() {
        return "Queen";
    }
}

File: chess-engine/src/main/java/model/pieces/Queen.kt
package model.pieces

import model.moves.Move
import model.util.Colour
import model.util.Offset

internal class Queen(colour: Colour) : DirectionPiece(colour) {

    override val directions: Array<Offset>
        get() = OFFSETS

    override val unicodeWhite: Int
        get() = 9813

    override val unicodeBlack: Int
        get() = 9819

    override val unsignedValue: Int
        get() = 8

    override val name: String
        get() = "Queen"

    override fun notifyMoveComplete(move: Move) {}

    override fun notifyMoveUndo(move: Move) {}

    companion object {
        private val OFFSETS = arrayOf(Offset.TOP_LEFT, Offset.UP, Offset.TOP_RIGHT, Offset.LEFT, Offset.RIGHT, Offset.BOTTOM_LEFT, Offset.DOWN, Offset.BOTTOM_RIGHT)
    }
}
